File: 733-FloodFill/733-FloodFill.java
// Last updated: 9/14/2025, 9:12:44 AM
class Solution {
    int [][]directions = new int[][]{
        {0,1},{1,0},{-1,0},{0,-1}
    };
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        //solving using bfs algorithm
        int n = image.length;
        int m = image[0].length;
        int initial = image[sr][sc];
        int [][]ans = image;
        Queue<int []> queue = new LinkedList<>();
        queue.add(new int[]{sr,sc});
        while(!queue.isEmpty()){
            int []indices = queue.poll();
            int x = indices[0];
            int y = indices[1];
            ans[x][y] = color;
            for(int []dir:directions){
                int nx = x + dir[0];
                int ny = y + dir[1];
                if(nx>=0 && ny>=0 && nx<n && ny<m && image[nx][ny]!=color && image[nx][ny]==initial){
                    queue.add(new int[]{nx,ny});
                }
            }
        }

        return ans;
    }
}

File: 733-FloodFill/733-FloodFill.py
# Last updated: 9/14/2025, 9:12:47 AM
class Solution:
    # DFS algorithm
    def color_change(self,image: List[List[int]], sr: int, sc: int,src: int, color: int):

        if(sr<0 or sr>=len(image) or sc<0 or sc>=len(image[0]) or image[sr][sc]!=src):
            return
        image[sr][sc]=color

        self.color_change(image,sr-1,sc,src,color)
        self.color_change(image,sr+1,sc,src,color)
        self.color_change(image,sr,sc-1,src,color)
        self.color_change(image,sr,sc+1,src,color)

    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        if image[sr][sc]==color:
            return image

        rows=len(image)
        cols=len(image[0])
        src=image[sr][sc]

        self.color_change(image,sr,sc,src,color)

        return image